frontend/src/metabase/components/FieldValuesWidget/FieldValuesWidget.tsx

```diff
@@ -170,41 +170,30 @@ export function FieldValuesWidgetInner({
     setLoadingState("LOADING");
     setOptions([]);
 
-    let newOptions: FieldValue[] = [];
-    let newValuesMode = valuesMode;
+    let values: FieldValue[] = [];
+    let has_more_values = false;
     try {
       if (canUseDashboardEndpoints(dashboard)) {
-        const { values, has_more_values } =
-          await dispatchFetchDashboardParameterValues(query);
-        newOptions = values;
-        newValuesMode = has_more_values ? "search" : newValuesMode;
+        ({ values, has_more_values } =
+          await dispatchFetchDashboardParameterValues(query));
       } else if (canUseCardEndpoints(question)) {
-        const { values, has_more_values } =
-          await dispatchFetchCardParameterValues(query);
-        newOptions = values;
-        newValuesMode = has_more_values ? "search" : newValuesMode;
+        ({ values, has_more_values } = await dispatchFetchCardParameterValues(
+          query,
+        ));
       } else if (canUseParameterEndpoints(parameter)) {
-        const { values, has_more_values } = await dispatchFetchParameterValues(
+        ({ values, has_more_values } = await dispatchFetchParameterValues(
           query,
-        );
-        newOptions = values;
-        newValuesMode = has_more_values ? "search" : newValuesMode;
+        ));
       } else {
-        newOptions = await fetchFieldValues(query);
-
-        newValuesMode = getValuesMode({
-          parameter,
-          fields,
-          disableSearch,
-          disablePKRemappingForSearch,
-        });
```

Comment:
This `getValuesMode` is unnecessary because the `valuesMode` is initialized with the exact same `getValuesMode` call.
https://github.com/metabase/metabase/blob/d96d020ec97bdca411f1a9f9bd45b6372c36a9bf/frontend/src/metabase/components/FieldValuesWidget/FieldValuesWidget.tsx#L147-L154